src/main/java/dan200/computercraft/shared/computer/core/InputState.java

```diff
@@ -93,6 +99,44 @@ public void mouseScroll( int direction, int x, int y )
         if( computer != null ) computer.mouseScroll( direction, x, y );
     }
 
+    @Override
+    public void mouseMove( int x, int y )
+    {
+        // Delegate mouse movement to the throttler, which will ensure the most
+        // recent mouse movement event is always received.
+
+        // An intentional decision here is to *not* update lastMouseX and lastMouseY,
+        // as the mouse_up event currently behaves based on the click and scroll
+        // events.
+
+        if( ComputerCraft.mouseMoveThrottle >= 0 && (lastMouseMoveX != x || lastMouseMoveY != y) )
+        {
+            lastMouseMoveX = x;
+            lastMouseMoveY = y;
+            mouseMoved = true; // The mouse_move event will be performed next time update() is called.
+        }
+    }
+
+    @Override
+    public void update()
+    {
+        // Handle mouse_move throttling:
+        if( mouseMoved && System.currentTimeMillis() - ComputerCraft.mouseMoveThrottle > lastMouseMove )
+        {
+            IComputer computer = owner.getComputer();
+            if( computer != null )
+            {
+                computer.mouseMove(
+                    lastMouseMoveX == -1 ? -1 : lastMouseMoveX + 1,
+                    lastMouseMoveY == -1 ? -1 : lastMouseMoveY + 1
+                );
+            }
+
+            mouseMoved = false;
+            lastMouseMove = System.currentTimeMillis();
```

Comment:
Sorry, meant to mention this yesterday and entirely forgot:

We should probably use some monotonic clock here. We don't want people unable to move the mouse because their system clock changes! Either `System.nanoTime` or the current tick<sup>1</sup> would be fine.

1: Obviously then it depends on tps, but I can live with that.